### PR TITLE
Switch over from dep to go modules

### PR DESCRIPTION
**Pull request by cblecker**
The generator wasn't picking up the root module's vendor folder. I don't know what/why/when. Easier to just convert over to go modules that I know works.

/assign @spiffxp @nikhita 
/hold
to ensure that the verify job works

### CODE REVIEW

**Comment by cblecker** [comment]
it works 🎉 
/hold cancel

File: hack/tools.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// +build tools
```
Comment (spiffxp) [inline hack/tools.go]: uh oh, a hack/ dir in community 😬 

I won't block on this, but for posterity's sake, why is this needed?

Comment (cblecker) [reply hack/tools.go]: The hack dir already exists for the CI scripts. What this file does is track the tools to vendor (recommended method, per https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module )

**Comment by nikhita** [comment]
Late to the party, but

/pony yay

**Comment by cblecker** [comment]
Oh, and I discovered why the previous dep configuration wasn't picking up the vendor folder: I cloned it outside of my GOPATH 🤦‍♂️ . Oh well, this works too lol.
